core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -1067,9 +1067,9 @@ private[spark] object Utils extends Logging {
     // finding the call site of a method.
     val SPARK_CORE_CLASS_REGEX =
       """^org\.apache\.spark(\.api\.java)?(\.util)?(\.rdd)?(\.broadcast)?\.[A-Z]""".r
-    val SCALA_CLASS_REGEX = """^scala""".r
+    val SCALA_CLASS = "scala"
```

Comment:
@srowen Thank you.

The reason for using a "lazy val" is as follows.

The function || takes two Boolean arguments, but only evaluates the second argument if the first is false:
```scala
scala> true || { println("!!"); false }
res0: Boolean = true
```

If "isSparkCoreClass" is true, "className.startsWith (SCALA_CORE_CLASS_PREFIX)" does not need to be evaluated.